### ISSUE TITLE
Spring Boot 2.2.0.RELEASE Beans form a cycle

### ISSUE THREAD

**Issue by kpetersson** [Playtika/feign-reactive#129]
Hey, I'm facing a problem when upgrading to Spring Boot 2.2.0.RELEASE
Error message:
```
Description:

The dependencies of some of the beans in the application context form a cycle:
.
.
.
.
      ↓
┌─────┐
|  reactiveFeignCloudBuilder defined in reactivefeign.spring.config.ReactiveFeignClientsConfiguration$ReactiveFeignClientsCloudConfiguration
└─────┘
```

**Comment by KuliginStepan** [comment]
@kpetersson I had the same error when upgrading to 2.2.0. As a workaround you can declare this bean in your `@Configuration` class:
```
@Bean
    @Scope("prototype")
    @ConditionalOnClass(WebReactiveFeign.class)
    public ReactiveFeignBuilder reactiveFeignBuilder(
        WebClient.Builder builder,
        @Autowired(required = false) WebClientFeignCustomizer webClientCustomizer) {
        return webClientCustomizer != null
            ? WebReactiveFeign.builder(builder, webClientCustomizer)
            : WebReactiveFeign.builder(builder);
    }
```
@kptfh @tdanylchuk When I went deeper into this issue, i found that this bean method defined in `reactivefeign.spring.config.ReactiveFeignClientsConfiguration` is not invoked. But when i removed  `@ConditionalOnClass(WebReactiveFeign.class)` all worked fine.

**Comment by tdanylchuk** [comment]
@KuliginStepan than you for your effort,
As for this issue we will investigate it from our side, most likely after spring cloud ```Hoxton```  will be released/

**Comment by kptfh** [comment]
fixed in 1.0.37 that was just released
It's based on
```
<spring-cloud-bom.version>Greenwich.SR3</spring-cloud-bom.version>
<spring-boot-dependencies-bom.version>2.2.1.RELEASE</spring-boot-dependencies-bom.version>
```
